spec/data-types-and-values.html

```diff
@@ -8,11 +8,50 @@ <h1>ECMAScript Language Types</h1>
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
       <p>TBD</p>
+
+      <emu-clause id="sec-recordtostring" aoid="RecordToString">
+        <h1>RecordToString ( _argument_ )</h1>
+        <p>The abstract operation RecordToString takes argument _argument_ (a Record) and convertes _argument_ to a String. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: Type(_argument_) is Record.
+            1. Return *"[record]"*.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <emu-clause id="sec-ecmascript-language-types-tuple-type">
       <h1>The Tuple Type</h1>
       <p>TBD</p>
+
+      <emu-clause id="sec-tupletostring" aoid="TupleToString">
+        <h1>TupleToString ( _argument_, [ _separator_, [ _mapper_ ] ] )</h1>
+        <p>The abstract operation TupleToString takes arguments _argument_, _separator_, and _mapper_ (a function object). _mapper_ will be invoked with *undefined* as the receiver. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: Type(_argument_) is Tuple.
+            1. Assert: _mapper_ is *undefined* or IsCallable(_mapper_) is *true*.
+            1. Assert: Type(_separator_) is Undefined or String.

```

Comment:
```suggestion
        <p>The abstract operation TupleToString takes arguments _argument_ (a Tuple), _separator_ (a String), and _mapper_ (a function object). _mapper_ will be invoked with *undefined* as the receiver. It performs the following steps when called:</p>
        <emu-alg>
            1. Assert: _mapper_ is *undefined* or IsCallable(_mapper_) is *true*.
```
i'm not sure the IsCallable assertion is needed either, given that the prose says "function object"